eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway.php

```diff
@@ -170,6 +170,26 @@ abstract class Gateway
      */
     abstract public function unHideSubtree($pathString);
 
+    /**
+     * @param string $pathString
+     **/
+    abstract public function setNodeWithChildrenInvisible(string $pathString): void;
+
+    /**
+     * @param string $pathString
+     **/
+    abstract public function setNodeWithChildrenVisible(string $pathString): void;
+
+    /**
+     * @param string $pathString
+     **/
+    abstract public function setNodeHidden(string $pathString): void;
+
+    /**
+     * @param string $pathString
+     **/
+    abstract public function setNodeUnhidden(string $pathString): void;
```

Comment:
Hmm, judging by implementation this is quite redundant so rather:
`toggleNodeVisibility(string $pathString, bool $visible): void;`

I haven't checked `setNodeWithChildrenVisible` and `setNodeWithChildrenInvisible` but maybe the same thing can be done there also?